### PR TITLE
Add leapp debug tools to initramfs upgrade module

### PR DESCRIPTION
**Pull request by matejmatuska**
Install a script with debug utilities to the Leapp upgrade dracut module.

### CODE REVIEW

**Comment by matejmatuska** [comment]
@vinzenz I believe this is what you had in mind :) ?

File: repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# library containing some useful functions for debugging in initramfs
+
+# mounts the sysroot
+leapp_dbg_mount() {
+    systemctl start sysroot.mount
```
Comment (vinzenz) [inline repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh:5]: you'd probably want it to be remounted rw

`mount -o remount,rw /sysroot

Comment (matejmatuska) [reply repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh:5]: I was thinking about it. Okay I will make it rw

File: repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+# source programs from $NEWROOT, mount if not mounted
+leapp_dbg_source() {
+    systemctl is-active sysroot.mount --quiet || {
+        echo "sysroot not mounted, mounting...";
+        leapp_dbg_mount || return 1
+    }
+    for dir in /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin; do
+        export PATH="$PATH:${NEWROOT}$dir"
```
Comment (vinzenz) [inline repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh:15]: /sysroot/bin and /sysroot/sbin is enough, you won't need /usr/{s,}bin nor /usr/local/{s,}bin

But you probably also want: `export LD_LIBRARY_PATH=/sysroot/lib64`

**Comment by matejmatuska** [comment]
One issue is that some programs added to `$PATH`, such as `file` won't work because they are missing shared libs.

File: repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    done
+}
+
+# chroot into $NEWROOT
+leapp_dbg_chroot() {
```
Comment (vinzenz) [inline repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh:20]: +1

File: repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh (hunk 2 above)
Comment (matejmatuska) [reply repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/90sys-upgrade/leapp_debug_tools.sh:15]: Yeah that's why the shared libs don't work

**Comment by matejmatuska** [comment]
Also I am thinking about whether the script should be sourced automatically?

**Comment by Rezney** [comment]
> Also I am thinking about whether the script should be sourced automatically?

@matejmatuska as discussed offline, go for the auto-sourcing.

**Comment by matejmatuska** [comment]
The tools script is now auto sourced. Tested in VM, works.

**Review by Rezney — APPROVED** [review]
Works like a charm. Well done!

**Pull request merged**